{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create files mapping FACS well and plate to basic allele metadata\n",
    "\n",
    "This script takes cherry picking and allele info files as input and produces a table relating FACS well and plate IDs to allele IDs and symbols. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "dualipa_inputs = \"../1_inputs\"\n",
    "dualipa_outputs = \"../3_outputs\"\n",
    "allele_collection_inputs = \"../../../1_allele_collection/1_inputs\"\n",
    "allele_collection_outputs = \"../../../1_allele_collection/3_outputs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orf_id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>nt_change</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_pos_38</th>\n",
       "      <th>spdi</th>\n",
       "      <th>ccsb_allele_id</th>\n",
       "      <th>ccsb_mutation_id</th>\n",
       "      <th>entry_plate</th>\n",
       "      <th>entry_well</th>\n",
       "      <th>collection</th>\n",
       "      <th>sequence</th>\n",
       "      <th>mut_type</th>\n",
       "      <th>mut_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>GBA</td>\n",
       "      <td>160G&gt;C</td>\n",
       "      <td>Val54Leu</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>1</td>\n",
       "      <td>155240033</td>\n",
       "      <td>NC_000001.11:155240033:C:G</td>\n",
       "      <td>ALE0000584</td>\n",
       "      <td>CCSBVarC000001</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>F12</td>\n",
       "      <td>RC4</td>\n",
       "      <td>ATGGAGTTTTCAAGTCCTTCCAGAGAGGAATGTCCCAAGCCTTTGA...</td>\n",
       "      <td>mutation_id_rc4</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>GBA</td>\n",
       "      <td>670C&gt;T</td>\n",
       "      <td>Leu224Phe</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>1</td>\n",
       "      <td>155238225</td>\n",
       "      <td>NC_000001.11:155238225:G:A</td>\n",
       "      <td>ALE00000002</td>\n",
       "      <td>CCSBVarC000002</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>C12</td>\n",
       "      <td>RC4</td>\n",
       "      <td>ATGGAGTTTTCAAGTCCTTCCAGAGAGGAATGTCCCAAGCCTTTGA...</td>\n",
       "      <td>mutation_id_rc4</td>\n",
       "      <td>73.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>GBA</td>\n",
       "      <td>887G&gt;A</td>\n",
       "      <td>Arg296Gln</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>1</td>\n",
       "      <td>155237453</td>\n",
       "      <td>NC_000001.11:155237453:C:T</td>\n",
       "      <td>ALE00000003</td>\n",
       "      <td>CCSBVarC000003</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>D12</td>\n",
       "      <td>RC4</td>\n",
       "      <td>ATGGAGTTTTCAAGTCCTTCCAGAGAGGAATGTCCCAAGCCTTTGA...</td>\n",
       "      <td>mutation_id_rc4</td>\n",
       "      <td>113.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>GBA</td>\n",
       "      <td>1448T&gt;C</td>\n",
       "      <td>Leu483Pro</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>1</td>\n",
       "      <td>155235252</td>\n",
       "      <td>NC_000001.11:155235252:A:G</td>\n",
       "      <td>ALE00000004</td>\n",
       "      <td>CCSBVarC000004</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>E12</td>\n",
       "      <td>RC4</td>\n",
       "      <td>ATGGAGTTTTCAAGTCCTTCCAGAGAGGAATGTCCCAAGCCTTTGA...</td>\n",
       "      <td>mutation_id_rc4</td>\n",
       "      <td>231.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>25</td>\n",
       "      <td>SLC9A3R1</td>\n",
       "      <td>328C&gt;G</td>\n",
       "      <td>Leu110Val</td>\n",
       "      <td>ENSG00000109062</td>\n",
       "      <td>17</td>\n",
       "      <td>74749174</td>\n",
       "      <td>NC_000017.11:74749174:C:G</td>\n",
       "      <td>ALE00000005</td>\n",
       "      <td>CCSBVarC000005</td>\n",
       "      <td>GDEhDisVCh_40045</td>\n",
       "      <td>E04</td>\n",
       "      <td>RC4</td>\n",
       "      <td>ATGAGCGCGGACGCAGCGGCCGGGGCGCCCCTGCCCCGGCTCTGCT...</td>\n",
       "      <td>mutation_id_rc4</td>\n",
       "      <td>252.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25459</th>\n",
       "      <td>8999</td>\n",
       "      <td>TTR</td>\n",
       "      <td>355G&gt;A</td>\n",
       "      <td>Asp119Asn</td>\n",
       "      <td>ENSG00000118271</td>\n",
       "      <td>18</td>\n",
       "      <td>31598586</td>\n",
       "      <td>NC_000018.10:31598586:G:A</td>\n",
       "      <td>ALE00008468</td>\n",
       "      <td>CCSBVarC008481</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Edgotyping3</td>\n",
       "      <td>ATGGCTTCTCATCGTCTGCTCCTCCTCTGCCTTGCTGGACTGGTAT...</td>\n",
       "      <td>mutation_id_edgotyping3</td>\n",
       "      <td>390.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25460</th>\n",
       "      <td>7323</td>\n",
       "      <td>PMP22</td>\n",
       "      <td>152A&gt;G</td>\n",
       "      <td>His51Arg</td>\n",
       "      <td>ENSG00000109099</td>\n",
       "      <td>17</td>\n",
       "      <td>15259120</td>\n",
       "      <td>NC_000017.11:15259120:T:C</td>\n",
       "      <td>ALE00008469</td>\n",
       "      <td>CCSBVarC008466</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Edgotyping3</td>\n",
       "      <td>ATGCTCCTCCTGTTGCTGAGTATCATCGTCCTCCACGTCGCGGTGC...</td>\n",
       "      <td>mutation_id_edgotyping3</td>\n",
       "      <td>394.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25461</th>\n",
       "      <td>2651</td>\n",
       "      <td>CLN6</td>\n",
       "      <td>89G&gt;C</td>\n",
       "      <td>Gly30Ala</td>\n",
       "      <td>ENSG00000128973</td>\n",
       "      <td>15</td>\n",
       "      <td>68218645</td>\n",
       "      <td>NC_000015.10:68218645:C:G</td>\n",
       "      <td>ALE00008470</td>\n",
       "      <td>CCSBVarC008436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Edgotyping3</td>\n",
       "      <td>ATGGAGGCGACGCGGAGGCGGCAGCACCTGGGAGCGACGGGCGGCC...</td>\n",
       "      <td>mutation_id_edgotyping3</td>\n",
       "      <td>436.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25462</th>\n",
       "      <td>7323</td>\n",
       "      <td>PMP22</td>\n",
       "      <td>403G&gt;A</td>\n",
       "      <td>Ala135Thr</td>\n",
       "      <td>ENSG00000109099</td>\n",
       "      <td>17</td>\n",
       "      <td>15230997</td>\n",
       "      <td>NC_000017.11:15230997:C:T</td>\n",
       "      <td>ALE00008471</td>\n",
       "      <td>CCSBVarC008467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Edgotyping3</td>\n",
       "      <td>ATGCTCCTCCTGTTGCTGAGTATCATCGTCCTCCACGTCGCGGTGC...</td>\n",
       "      <td>mutation_id_edgotyping3</td>\n",
       "      <td>1840.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25463</th>\n",
       "      <td>7215</td>\n",
       "      <td>NR0B1</td>\n",
       "      <td>376G&gt;A</td>\n",
       "      <td>Val126Met</td>\n",
       "      <td>ENSG00000169297</td>\n",
       "      <td>X</td>\n",
       "      <td>30308988</td>\n",
       "      <td>NC_000023.11:30308988:C:T</td>\n",
       "      <td>ALE00008472</td>\n",
       "      <td>CCSBVarC008465</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Edgotyping3</td>\n",
       "      <td>ATGGCGGGCGAGAACCACCAGTGGCAGGGCAGCATCCTCTACAACA...</td>\n",
       "      <td>mutation_id_edgotyping3</td>\n",
       "      <td>1976.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25464 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       orf_id    symbol nt_change  aa_change  ensembl_gene_id chr  chr_pos_38  \\\n",
       "0           2       GBA    160G>C   Val54Leu  ENSG00000177628   1   155240033   \n",
       "1           2       GBA    670C>T  Leu224Phe  ENSG00000177628   1   155238225   \n",
       "2           2       GBA    887G>A  Arg296Gln  ENSG00000177628   1   155237453   \n",
       "3           2       GBA   1448T>C  Leu483Pro  ENSG00000177628   1   155235252   \n",
       "4          25  SLC9A3R1    328C>G  Leu110Val  ENSG00000109062  17    74749174   \n",
       "...       ...       ...       ...        ...              ...  ..         ...   \n",
       "25459    8999       TTR    355G>A  Asp119Asn  ENSG00000118271  18    31598586   \n",
       "25460    7323     PMP22    152A>G   His51Arg  ENSG00000109099  17    15259120   \n",
       "25461    2651      CLN6     89G>C   Gly30Ala  ENSG00000128973  15    68218645   \n",
       "25462    7323     PMP22    403G>A  Ala135Thr  ENSG00000109099  17    15230997   \n",
       "25463    7215     NR0B1    376G>A  Val126Met  ENSG00000169297   X    30308988   \n",
       "\n",
       "                             spdi ccsb_allele_id ccsb_mutation_id  \\\n",
       "0      NC_000001.11:155240033:C:G     ALE0000584   CCSBVarC000001   \n",
       "1      NC_000001.11:155238225:G:A    ALE00000002   CCSBVarC000002   \n",
       "2      NC_000001.11:155237453:C:T    ALE00000003   CCSBVarC000003   \n",
       "3      NC_000001.11:155235252:A:G    ALE00000004   CCSBVarC000004   \n",
       "4       NC_000017.11:74749174:C:G    ALE00000005   CCSBVarC000005   \n",
       "...                           ...            ...              ...   \n",
       "25459   NC_000018.10:31598586:G:A    ALE00008468   CCSBVarC008481   \n",
       "25460   NC_000017.11:15259120:T:C    ALE00008469   CCSBVarC008466   \n",
       "25461   NC_000015.10:68218645:C:G    ALE00008470   CCSBVarC008436   \n",
       "25462   NC_000017.11:15230997:C:T    ALE00008471   CCSBVarC008467   \n",
       "25463   NC_000023.11:30308988:C:T    ALE00008472   CCSBVarC008465   \n",
       "\n",
       "            entry_plate entry_well   collection  \\\n",
       "0      GDEhDisVCh_40054        F12          RC4   \n",
       "1      GDEhDisVCh_40054        C12          RC4   \n",
       "2      GDEhDisVCh_40054        D12          RC4   \n",
       "3      GDEhDisVCh_40054        E12          RC4   \n",
       "4      GDEhDisVCh_40045        E04          RC4   \n",
       "...                 ...        ...          ...   \n",
       "25459               NaN        NaN  Edgotyping3   \n",
       "25460               NaN        NaN  Edgotyping3   \n",
       "25461               NaN        NaN  Edgotyping3   \n",
       "25462               NaN        NaN  Edgotyping3   \n",
       "25463               NaN        NaN  Edgotyping3   \n",
       "\n",
       "                                                sequence  \\\n",
       "0      ATGGAGTTTTCAAGTCCTTCCAGAGAGGAATGTCCCAAGCCTTTGA...   \n",
       "1      ATGGAGTTTTCAAGTCCTTCCAGAGAGGAATGTCCCAAGCCTTTGA...   \n",
       "2      ATGGAGTTTTCAAGTCCTTCCAGAGAGGAATGTCCCAAGCCTTTGA...   \n",
       "3      ATGGAGTTTTCAAGTCCTTCCAGAGAGGAATGTCCCAAGCCTTTGA...   \n",
       "4      ATGAGCGCGGACGCAGCGGCCGGGGCGCCCCTGCCCCGGCTCTGCT...   \n",
       "...                                                  ...   \n",
       "25459  ATGGCTTCTCATCGTCTGCTCCTCCTCTGCCTTGCTGGACTGGTAT...   \n",
       "25460  ATGCTCCTCCTGTTGCTGAGTATCATCGTCCTCCACGTCGCGGTGC...   \n",
       "25461  ATGGAGGCGACGCGGAGGCGGCAGCACCTGGGAGCGACGGGCGGCC...   \n",
       "25462  ATGCTCCTCCTGTTGCTGAGTATCATCGTCCTCCACGTCGCGGTGC...   \n",
       "25463  ATGGCGGGCGAGAACCACCAGTGGCAGGGCAGCATCCTCTACAACA...   \n",
       "\n",
       "                      mut_type  mut_id  \n",
       "0              mutation_id_rc4     6.0  \n",
       "1              mutation_id_rc4    73.0  \n",
       "2              mutation_id_rc4   113.0  \n",
       "3              mutation_id_rc4   231.0  \n",
       "4              mutation_id_rc4   252.0  \n",
       "...                        ...     ...  \n",
       "25459  mutation_id_edgotyping3   390.0  \n",
       "25460  mutation_id_edgotyping3   394.0  \n",
       "25461  mutation_id_edgotyping3   436.0  \n",
       "25462  mutation_id_edgotyping3  1840.0  \n",
       "25463  mutation_id_edgotyping3  1976.0  \n",
       "\n",
       "[25464 rows x 16 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get allele metadata\n",
    "allele_info_df = pd.read_table(f\"{allele_collection_inputs}/VarChamp_Consolidated_allele_collection.tsv\")\n",
    "other_cols = [i for i in allele_info_df.columns if \"mutation_id_\" not in i]\n",
    "mut_cols = [i for i in allele_info_df.columns if \"mutation_id_\" in i]\n",
    "\n",
    "allele_info_df = allele_info_df.melt(id_vars=other_cols, \n",
    "                                     value_vars=mut_cols, \n",
    "                                     var_name='mut_type', \n",
    "                                     value_name='mut_id')\n",
    "allele_info_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get gene-level metadata\n",
    "gene_info_df = pd.read_csv(f\"{allele_collection_outputs}/orf_info.csv\")\n",
    "gene_info_df = gene_info_df[['orf_id', 'symbol']]\n",
    "orf_to_symbol = gene_info_df.set_index(\"orf_id\")[\"symbol\"].to_dict()\n",
    "orf_to_symbol = {str(k): v for k, v in orf_to_symbol.items()}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Vidal lab ran experiment once and analyzed the data. Some alleles didn't pass the qa/qc thresholds or got no data. Batch5 is is a re-run for all alleles that failed in the first set of batches. Here, we process everything together from scratch. An allele should always be compared to the wt on the same batch and plate. The summary stats can then be compared across batches. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orf_id</th>\n",
       "      <th>mut_id</th>\n",
       "      <th>node_type</th>\n",
       "      <th>dest_pla_id</th>\n",
       "      <th>dest_pla</th>\n",
       "      <th>dest_well</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>56360.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>A12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>56360.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>B12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>56360.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>C12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>56360.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>D12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>9905.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>F02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>20</td>\n",
       "      <td>VUSDUALXP_20</td>\n",
       "      <td>A08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>21</td>\n",
       "      <td>VUSDUALXP_21</td>\n",
       "      <td>A07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>21</td>\n",
       "      <td>VUSDUALXP_21</td>\n",
       "      <td>A08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>22</td>\n",
       "      <td>VUSDUALXP_22</td>\n",
       "      <td>A07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>22</td>\n",
       "      <td>VUSDUALXP_22</td>\n",
       "      <td>A08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2311 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      orf_id  mut_id    node_type  dest_pla_id      dest_pla dest_well\n",
       "78   56360.0     0.0           wt           23  VUSDUALXP_23       A12\n",
       "79   56360.0     0.0           wt           23  VUSDUALXP_23       B12\n",
       "80   56360.0     0.0           wt           23  VUSDUALXP_23       C12\n",
       "81   56360.0     0.0           wt           23  VUSDUALXP_23       D12\n",
       "10    9905.0     0.0           wt           23  VUSDUALXP_23       F02\n",
       "..       ...     ...          ...          ...           ...       ...\n",
       "119      NaN     NaN  pdest_empty           20  VUSDUALXP_20       A08\n",
       "124      NaN     NaN  pdest_empty           21  VUSDUALXP_21       A07\n",
       "125      NaN     NaN  pdest_empty           21  VUSDUALXP_21       A08\n",
       "130      NaN     NaN  pdest_empty           22  VUSDUALXP_22       A07\n",
       "131      NaN     NaN  pdest_empty           22  VUSDUALXP_22       A08\n",
       "\n",
       "[2311 rows x 6 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in cherry picking files for both the initial batches and the second round\n",
    "cp_col = ['orf_id', 'mut_id', 'node_type','dest_pla_id', 'dest_pla', 'dest_well']\n",
    "\n",
    "pdest_layout_initial_batches_df = pd.read_csv(f'{dualipa_inputs}/cherry_picking_files/summary_CP_VUSDUALXP.csv', index_col=0)[cp_col]\n",
    "pdest_layout_b5_df = pd.read_csv(f'{dualipa_inputs}/cherry_picking_files/summary_CP_VUSDUALXP_batch5.csv', index_col=0)[cp_col]\n",
    "\n",
    "pdest_layout_df = pd.concat([pdest_layout_b5_df, pdest_layout_initial_batches_df])\n",
    "pdest_layout_df\n",
    "### This doesn't have well A01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>code</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>perfectly validated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>partially validated, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>wild type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>partial wild type, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>target and off-target mutation, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>off-target mutation, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>truncated, &lt; 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>99</td>\n",
       "      <td>no reads</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   code                                      description\n",
       "0     1                              perfectly validated\n",
       "1     2             partially validated, >= 50% coverage\n",
       "2     3                                        wild type\n",
       "3     4               partial wild type, >= 50% coverage\n",
       "4     5  target and off-target mutation, >= 50% coverage\n",
       "5     6             off-target mutation, >= 50% coverage\n",
       "6     7                        truncated, < 50% coverage\n",
       "7    99                                         no reads"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "seq_confirm_code_df = pd.read_table(f\"{allele_collection_inputs}/sequence_confirmation_class_code.tsv\")\n",
    "with pd.option_context('display.max_columns', None):\n",
    "    display(seq_confirm_code_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>symbol</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>orf_id_wt</th>\n",
       "      <th>mutation_id_old</th>\n",
       "      <th>ccsb_mutation_id</th>\n",
       "      <th>ccsb_allele_id</th>\n",
       "      <th>spdi</th>\n",
       "      <th>nt_change</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>collection</th>\n",
       "      <th>entry_plate_orig</th>\n",
       "      <th>entry_well_orig</th>\n",
       "      <th>entry_plate_conso</th>\n",
       "      <th>entry_well_conso</th>\n",
       "      <th>entry_seq_pool</th>\n",
       "      <th>db_plate</th>\n",
       "      <th>db_well</th>\n",
       "      <th>n2h_plate</th>\n",
       "      <th>n2h_well</th>\n",
       "      <th>dualip_plate</th>\n",
       "      <th>dualip_well</th>\n",
       "      <th>mislocalization_plate</th>\n",
       "      <th>mislocalization_well</th>\n",
       "      <th>entry_sequenced</th>\n",
       "      <th>entry_sequence_confirmation_class</th>\n",
       "      <th>db_sequenced</th>\n",
       "      <th>db_sequence_confirmation_class</th>\n",
       "      <th>n2h_sequenced</th>\n",
       "      <th>n2h_sequence_confirmation_class</th>\n",
       "      <th>dualip_sequenced</th>\n",
       "      <th>dualip_sequence_confirmation_class</th>\n",
       "      <th>mislocalization_sequenced</th>\n",
       "      <th>mislocalization_sequence_confirmation_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>CCSBVarC000001</td>\n",
       "      <td>ALE0000584</td>\n",
       "      <td>NC_000001.11:155240033:C:G</td>\n",
       "      <td>160G&gt;C</td>\n",
       "      <td>Val54Leu</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>H01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>F12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>H01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>73</td>\n",
       "      <td>CCSBVarC000002</td>\n",
       "      <td>ALE00000002</td>\n",
       "      <td>NC_000001.11:155238225:G:A</td>\n",
       "      <td>670C&gt;T</td>\n",
       "      <td>Leu224Phe</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>E01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>C12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>E01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>113</td>\n",
       "      <td>CCSBVarC000003</td>\n",
       "      <td>ALE00000003</td>\n",
       "      <td>NC_000001.11:155237453:C:T</td>\n",
       "      <td>887G&gt;A</td>\n",
       "      <td>Arg296Gln</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>F01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>D12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>F01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>231</td>\n",
       "      <td>CCSBVarC000004</td>\n",
       "      <td>ALE00000004</td>\n",
       "      <td>NC_000001.11:155235252:A:G</td>\n",
       "      <td>1448T&gt;C</td>\n",
       "      <td>Leu483Pro</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>G01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>E12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>G01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>ENSG00000177628</td>\n",
       "      <td>2</td>\n",
       "      <td>213510</td>\n",
       "      <td>CCSBVarC003869</td>\n",
       "      <td>ALE00003869</td>\n",
       "      <td>NC_000001.11:155239934:G:A</td>\n",
       "      <td>259C&gt;T</td>\n",
       "      <td>Arg87Trp</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1035</td>\n",
       "      <td>B03</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>B02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CegsMutGDDh1035</td>\n",
       "      <td>B03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8705</th>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>ENSG00000168036</td>\n",
       "      <td>100070227</td>\n",
       "      <td>212487</td>\n",
       "      <td>CCSBVarC007073</td>\n",
       "      <td>ALE00007071</td>\n",
       "      <td>NC_000003.12:41225785:A:G</td>\n",
       "      <td>860A&gt;G</td>\n",
       "      <td>Asn287Ser</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>B07</td>\n",
       "      <td>GDEhDisVCh_40034</td>\n",
       "      <td>E09</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8706</th>\n",
       "      <td>TBX19</td>\n",
       "      <td>ENSG00000143178</td>\n",
       "      <td>100070258</td>\n",
       "      <td>212488</td>\n",
       "      <td>CCSBVarC007074</td>\n",
       "      <td>ALE00007072</td>\n",
       "      <td>NC_000001.11:168291339:C:T</td>\n",
       "      <td>383C&gt;T</td>\n",
       "      <td>Ser128Phe</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1043</td>\n",
       "      <td>A07</td>\n",
       "      <td>GDEhDisVCh_40046</td>\n",
       "      <td>B02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CegsMutGDDh1043</td>\n",
       "      <td>A07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8707</th>\n",
       "      <td>TBX19</td>\n",
       "      <td>ENSG00000143178</td>\n",
       "      <td>100070258</td>\n",
       "      <td>212489</td>\n",
       "      <td>CCSBVarC007075</td>\n",
       "      <td>ALE00007073</td>\n",
       "      <td>NC_000001.11:168291213:T:G</td>\n",
       "      <td>257T&gt;G</td>\n",
       "      <td>Met86Arg</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1043</td>\n",
       "      <td>G08</td>\n",
       "      <td>GDEhDisVCh_40046</td>\n",
       "      <td>F02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CegsMutGDDh1043</td>\n",
       "      <td>G08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8708</th>\n",
       "      <td>GDF2</td>\n",
       "      <td>ENSG00000263761</td>\n",
       "      <td>100070273</td>\n",
       "      <td>212491</td>\n",
       "      <td>CCSBVarC007076</td>\n",
       "      <td>ALE00007074</td>\n",
       "      <td>NC_000010.11:47322871:G:T</td>\n",
       "      <td>203G&gt;T</td>\n",
       "      <td>Arg68Leu</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>E04</td>\n",
       "      <td>GDEhDisVCh_40033</td>\n",
       "      <td>B08</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8709</th>\n",
       "      <td>GDF2</td>\n",
       "      <td>ENSG00000263761</td>\n",
       "      <td>100070273</td>\n",
       "      <td>212492</td>\n",
       "      <td>CCSBVarC007077</td>\n",
       "      <td>ALE00007075</td>\n",
       "      <td>NC_000010.11:47325491:C:T</td>\n",
       "      <td>997C&gt;T</td>\n",
       "      <td>Arg333Trp</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>B06</td>\n",
       "      <td>GDEhDisVCh_40033</td>\n",
       "      <td>E08</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8710 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      symbol  ensembl_gene_id  orf_id_wt  mutation_id_old ccsb_mutation_id  \\\n",
       "0       GBA1  ENSG00000177628          2                6   CCSBVarC000001   \n",
       "1       GBA1  ENSG00000177628          2               73   CCSBVarC000002   \n",
       "2       GBA1  ENSG00000177628          2              113   CCSBVarC000003   \n",
       "3       GBA1  ENSG00000177628          2              231   CCSBVarC000004   \n",
       "4       GBA1  ENSG00000177628          2           213510   CCSBVarC003869   \n",
       "...      ...              ...        ...              ...              ...   \n",
       "8705  CTNNB1  ENSG00000168036  100070227           212487   CCSBVarC007073   \n",
       "8706   TBX19  ENSG00000143178  100070258           212488   CCSBVarC007074   \n",
       "8707   TBX19  ENSG00000143178  100070258           212489   CCSBVarC007075   \n",
       "8708    GDF2  ENSG00000263761  100070273           212491   CCSBVarC007076   \n",
       "8709    GDF2  ENSG00000263761  100070273           212492   CCSBVarC007077   \n",
       "\n",
       "     ccsb_allele_id                        spdi nt_change  aa_change  \\\n",
       "0        ALE0000584  NC_000001.11:155240033:C:G    160G>C   Val54Leu   \n",
       "1       ALE00000002  NC_000001.11:155238225:G:A    670C>T  Leu224Phe   \n",
       "2       ALE00000003  NC_000001.11:155237453:C:T    887G>A  Arg296Gln   \n",
       "3       ALE00000004  NC_000001.11:155235252:A:G   1448T>C  Leu483Pro   \n",
       "4       ALE00003869  NC_000001.11:155239934:G:A    259C>T   Arg87Trp   \n",
       "...             ...                         ...       ...        ...   \n",
       "8705    ALE00007071   NC_000003.12:41225785:A:G    860A>G  Asn287Ser   \n",
       "8706    ALE00007072  NC_000001.11:168291339:C:T    383C>T  Ser128Phe   \n",
       "8707    ALE00007073  NC_000001.11:168291213:T:G    257T>G   Met86Arg   \n",
       "8708    ALE00007074   NC_000010.11:47322871:G:T    203G>T   Arg68Leu   \n",
       "8709    ALE00007075   NC_000010.11:47325491:C:T    997C>T  Arg333Trp   \n",
       "\n",
       "     collection  entry_plate_orig entry_well_orig entry_plate_conso  \\\n",
       "0           RC4  RC4_Mut_GDEh1026             H01  GDEhDisVCh_40054   \n",
       "1           RC4  RC4_Mut_GDEh1026             E01  GDEhDisVCh_40054   \n",
       "2           RC4  RC4_Mut_GDEh1026             F01  GDEhDisVCh_40054   \n",
       "3           RC4  RC4_Mut_GDEh1026             G01  GDEhDisVCh_40054   \n",
       "4         CEGS2   CegsMutGDEh1035             B03  GDEhDisVCh_40054   \n",
       "...         ...               ...             ...               ...   \n",
       "8705      CEGS2   CegsMutGDEh1023             B07  GDEhDisVCh_40034   \n",
       "8706      CEGS2   CegsMutGDEh1043             A07  GDEhDisVCh_40046   \n",
       "8707      CEGS2   CegsMutGDEh1043             G08  GDEhDisVCh_40046   \n",
       "8708      CEGS2   CegsMutGDEh1023             E04  GDEhDisVCh_40033   \n",
       "8709      CEGS2   CegsMutGDEh1023             B06  GDEhDisVCh_40033   \n",
       "\n",
       "     entry_well_conso  entry_seq_pool          db_plate db_well n2h_plate  \\\n",
       "0                 F12             2.0  RC4_Mut_GDDh1026     H01       NaN   \n",
       "1                 C12             2.0  RC4_Mut_GDDh1026     E01       NaN   \n",
       "2                 D12             2.0  RC4_Mut_GDDh1026     F01       NaN   \n",
       "3                 E12             2.0  RC4_Mut_GDDh1026     G01       NaN   \n",
       "4                 B02             2.0   CegsMutGDDh1035     B03       NaN   \n",
       "...               ...             ...               ...     ...       ...   \n",
       "8705              E09             2.0               NaN     NaN       NaN   \n",
       "8706              B02             2.0   CegsMutGDDh1043     A07       NaN   \n",
       "8707              F02             2.0   CegsMutGDDh1043     G08       NaN   \n",
       "8708              B08             1.0               NaN     NaN       NaN   \n",
       "8709              E08             1.0               NaN     NaN       NaN   \n",
       "\n",
       "     n2h_well dualip_plate dualip_well mislocalization_plate  \\\n",
       "0         NaN          NaN         NaN                   NaN   \n",
       "1         NaN          NaN         NaN                   NaN   \n",
       "2         NaN          NaN         NaN                   NaN   \n",
       "3         NaN          NaN         NaN                   NaN   \n",
       "4         NaN          NaN         NaN                   NaN   \n",
       "...       ...          ...         ...                   ...   \n",
       "8705      NaN          NaN         NaN                   NaN   \n",
       "8706      NaN          NaN         NaN                   NaN   \n",
       "8707      NaN          NaN         NaN                   NaN   \n",
       "8708      NaN          NaN         NaN                   NaN   \n",
       "8709      NaN          NaN         NaN                   NaN   \n",
       "\n",
       "     mislocalization_well  entry_sequenced  entry_sequence_confirmation_class  \\\n",
       "0                     NaN                1                                1.0   \n",
       "1                     NaN                1                                1.0   \n",
       "2                     NaN                1                                1.0   \n",
       "3                     NaN                1                                1.0   \n",
       "4                     NaN                1                                1.0   \n",
       "...                   ...              ...                                ...   \n",
       "8705                  NaN                1                                1.0   \n",
       "8706                  NaN                1                                7.0   \n",
       "8707                  NaN                1                                1.0   \n",
       "8708                  NaN                1                                5.0   \n",
       "8709                  NaN                1                                5.0   \n",
       "\n",
       "      db_sequenced  db_sequence_confirmation_class  n2h_sequenced  \\\n",
       "0                1                             2.0              0   \n",
       "1                1                             1.0              0   \n",
       "2                1                             7.0              0   \n",
       "3                1                             2.0              0   \n",
       "4                1                             2.0              0   \n",
       "...            ...                             ...            ...   \n",
       "8705             0                             NaN              0   \n",
       "8706             1                             7.0              0   \n",
       "8707             1                             1.0              0   \n",
       "8708             0                             NaN              0   \n",
       "8709             0                             NaN              0   \n",
       "\n",
       "      n2h_sequence_confirmation_class  dualip_sequenced  \\\n",
       "0                                 NaN                 0   \n",
       "1                                 NaN                 0   \n",
       "2                                 NaN                 0   \n",
       "3                                 NaN                 0   \n",
       "4                                 NaN                 0   \n",
       "...                               ...               ...   \n",
       "8705                              NaN                 0   \n",
       "8706                              NaN                 0   \n",
       "8707                              NaN                 0   \n",
       "8708                              NaN                 0   \n",
       "8709                              NaN                 0   \n",
       "\n",
       "      dualip_sequence_confirmation_class  mislocalization_sequenced  \\\n",
       "0                                    NaN                          0   \n",
       "1                                    NaN                          0   \n",
       "2                                    NaN                          0   \n",
       "3                                    NaN                          0   \n",
       "4                                    NaN                          0   \n",
       "...                                  ...                        ...   \n",
       "8705                                 NaN                          0   \n",
       "8706                                 NaN                          0   \n",
       "8707                                 NaN                          0   \n",
       "8708                                 NaN                          0   \n",
       "8709                                 NaN                          0   \n",
       "\n",
       "      mislocalization_sequence_confirmation_class  \n",
       "0                                             NaN  \n",
       "1                                             NaN  \n",
       "2                                             NaN  \n",
       "3                                             NaN  \n",
       "4                                             NaN  \n",
       "...                                           ...  \n",
       "8705                                          NaN  \n",
       "8706                                          NaN  \n",
       "8707                                          NaN  \n",
       "8708                                          NaN  \n",
       "8709                                          NaN  \n",
       "\n",
       "[8710 rows x 33 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "array([nan, 'VUSMutpDEST2_01', 'VUSMutpDEST2_02', 'VUSMutpDEST2_03',\n",
       "       'VUSMutpDEST2_04', 'VUSMutpDEST2_05', 'VUSMutpDEST2_06',\n",
       "       'VUSMutpDEST2_07', 'VUSMutpDEST2_08', 'VUSMutpDEST2_09',\n",
       "       'VUSMutpDEST2_11', 'VUSMutpDEST2_12', 'VUSMutpDEST2_13',\n",
       "       'VUSMutpDEST2_14', 'VUSMutpDEST2_15', 'VUSMutpDEST2_16'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_confirm_res_df = pd.read_table(f\"{allele_collection_inputs}/VarChampSeqConfirmationResult.tsv\")\n",
    "with pd.option_context('display.max_columns', None):\n",
    "    display(seq_confirm_res_df)\n",
    "seq_confirm_res_df[\"dualip_plate\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[nan  1.  5.  6.  4.  3.  7. 99.  2.]\n"
     ]
    }
   ],
   "source": [
    "pdest_layout_seq_conf_df = pd.merge(pdest_layout_df, \n",
    "        seq_confirm_res_df[\n",
    "        [\"symbol\",\"orf_id_wt\",\"mutation_id_old\",\"spdi\",\"dualip_sequenced\",\"dualip_sequence_confirmation_class\",\"aa_change\",\"nt_change\",\"ccsb_mutation_id\"]\n",
    "        ].rename({\n",
    "            \"orf_id_wt\": \"orf_id\",\n",
    "            \"mutation_id_old\": \"mut_id\"\n",
    "        }, axis=1), \n",
    "    on=[\"orf_id\",\"mut_id\"],\n",
    "    how=\"left\"\n",
    ")\n",
    "print(pdest_layout_seq_conf_df[\"dualip_sequence_confirmation_class\"].unique())\n",
    "\n",
    "## QC with allele collection df for consistency\n",
    "meta_cols = ['orf_id', 'mut_id', 'aa_change', 'nt_change', 'symbol']\n",
    "pdest_layout_seq_conf_df = pdest_layout_seq_conf_df.merge(allele_info_df[meta_cols], on=['orf_id', 'mut_id'], how='left', suffixes=(\"\", \"_ac\"))\n",
    "pdest_layout_seq_conf_df\n",
    "\n",
    "assert(all(pdest_layout_seq_conf_df.dropna(subset=\"symbol\")[\"symbol\"]==pdest_layout_seq_conf_df.dropna(subset=\"symbol\")[\"symbol_ac\"]))\n",
    "assert(all(pdest_layout_seq_conf_df.dropna(subset=\"aa_change\")[\"aa_change\"]==pdest_layout_seq_conf_df.dropna(subset=\"aa_change\")[\"aa_change_ac\"]))\n",
    "assert(all(pdest_layout_seq_conf_df.dropna(subset=\"nt_change\")[\"nt_change\"]==pdest_layout_seq_conf_df.dropna(subset=\"nt_change\")[\"nt_change_ac\"]))\n",
    "pdest_layout_seq_conf_df = pdest_layout_seq_conf_df.drop(columns=[\"symbol_ac\",\"aa_change_ac\",\"nt_change_ac\"])\n",
    "\n",
    "## replace NAs\n",
    "pdest_layout_seq_conf_df[\"orf_id\"] = pdest_layout_seq_conf_df[\"orf_id\"].astype(\"Int64\").astype(str)\n",
    "pdest_layout_seq_conf_df[\"orf_id\"] = pdest_layout_seq_conf_df[\"orf_id\"].replace(\"<NA>\", np.nan)\n",
    "pdest_layout_seq_conf_df\n",
    "\n",
    "## Replace missing symbols\n",
    "pdest_layout_seq_conf_df[[\"aa_change\", \"nt_change\"]] = pdest_layout_seq_conf_df[[\"aa_change\", \"nt_change\"]].fillna(\"WT\")\n",
    "pdest_layout_seq_conf_df[\"symbol\"] = pdest_layout_seq_conf_df[\"symbol\"].fillna(pdest_layout_seq_conf_df[\"orf_id\"].map(orf_to_symbol))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdest_layout_seq_conf_df.to_csv(f'{dualipa_outputs}/dualipa_experimental_layout.csv', index=None)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "varchamp",
   "language": "python",
   "name": "varchamp"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
