{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create files mapping FACS well and plate to basic allele metadata\n",
    "\n",
    "This script takes cherry picking and allele info files as input and produces a table relating FACS well and plate IDs to allele IDs and symbols. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "dualipa_inputs = \"../1_inputs\"\n",
    "dualipa_outputs = \"../3_outputs\"\n",
    "allele_collection_inputs = \"../../../1_allele_collection/1_inputs\"\n",
    "allele_collection_outputs = \"../../../1_allele_collection/3_outputs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get allele metadata\n",
    "allele_info_df = pd.read_table(f\"{allele_collection_inputs}/VarChamp_Consolidated_allele_collection.tsv\")\n",
    "other_cols = [i for i in allele_info_df.columns if \"mutation_id_\" not in i]\n",
    "mut_cols = [i for i in allele_info_df.columns if \"mutation_id_\" in i]\n",
    "\n",
    "allele_info_df = allele_info_df.melt(id_vars=other_cols, \n",
    "                                     value_vars=mut_cols, \n",
    "                                     var_name='mut_type', \n",
    "                                     value_name='mut_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get gene-level metadata\n",
    "gene_info_df = pd.read_csv(f\"{allele_collection_outputs}/orf_info.csv\")\n",
    "gene_info_df = gene_info_df[['orf_id', 'symbol']]\n",
    "orf_to_symbol = gene_info_df.set_index(\"orf_id\")[\"symbol\"].to_dict()\n",
    "orf_to_symbol = {str(k): v for k, v in orf_to_symbol.items()}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Vidal lab ran experiment once and analyzed the data. Some alleles didn't pass the qa/qc thresholds or got no data. Batch5 is is a re-run for all alleles that failed in the first set of batches. Here, we process everything together from scratch. An allele should always be compared to the wt on the same batch and plate. The summary stats can then be compared across batches. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orf_id</th>\n",
       "      <th>mut_id</th>\n",
       "      <th>node_type</th>\n",
       "      <th>dest_pla_id</th>\n",
       "      <th>dest_pla</th>\n",
       "      <th>dest_well</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>56360.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>A12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>56360.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>B12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>56360.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>C12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>56360.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>D12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>9905.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>wt</td>\n",
       "      <td>23</td>\n",
       "      <td>VUSDUALXP_23</td>\n",
       "      <td>F02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>20</td>\n",
       "      <td>VUSDUALXP_20</td>\n",
       "      <td>A08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>21</td>\n",
       "      <td>VUSDUALXP_21</td>\n",
       "      <td>A07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>21</td>\n",
       "      <td>VUSDUALXP_21</td>\n",
       "      <td>A08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>22</td>\n",
       "      <td>VUSDUALXP_22</td>\n",
       "      <td>A07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pdest_empty</td>\n",
       "      <td>22</td>\n",
       "      <td>VUSDUALXP_22</td>\n",
       "      <td>A08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2311 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      orf_id  mut_id    node_type  dest_pla_id      dest_pla dest_well\n",
       "78   56360.0     0.0           wt           23  VUSDUALXP_23       A12\n",
       "79   56360.0     0.0           wt           23  VUSDUALXP_23       B12\n",
       "80   56360.0     0.0           wt           23  VUSDUALXP_23       C12\n",
       "81   56360.0     0.0           wt           23  VUSDUALXP_23       D12\n",
       "10    9905.0     0.0           wt           23  VUSDUALXP_23       F02\n",
       "..       ...     ...          ...          ...           ...       ...\n",
       "119      NaN     NaN  pdest_empty           20  VUSDUALXP_20       A08\n",
       "124      NaN     NaN  pdest_empty           21  VUSDUALXP_21       A07\n",
       "125      NaN     NaN  pdest_empty           21  VUSDUALXP_21       A08\n",
       "130      NaN     NaN  pdest_empty           22  VUSDUALXP_22       A07\n",
       "131      NaN     NaN  pdest_empty           22  VUSDUALXP_22       A08\n",
       "\n",
       "[2311 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in cherry picking files for both the initial batches and the second round\n",
    "cp_col = ['orf_id', 'mut_id', 'node_type','dest_pla_id', 'dest_pla', 'dest_well']\n",
    "\n",
    "pdest_layout_initial_batches_df = pd.read_csv(f'{dualipa_inputs}/cherry_picking_files/summary_CP_VUSDUALXP.csv', index_col=0)[cp_col]\n",
    "pdest_layout_b5_df = pd.read_csv(f'{dualipa_inputs}/cherry_picking_files/summary_CP_VUSDUALXP_batch5.csv', index_col=0)[cp_col]\n",
    "\n",
    "pdest_layout_df = pd.concat([pdest_layout_b5_df, pdest_layout_initial_batches_df])\n",
    "pdest_layout_df\n",
    "### This doesn't have well A01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>code</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>perfectly validated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>partially validated, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>wild type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>partial wild type, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>target and off-target mutation, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>off-target mutation, &gt;= 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>truncated, &lt; 50% coverage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>99</td>\n",
       "      <td>no reads</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   code                                      description\n",
       "0     1                              perfectly validated\n",
       "1     2             partially validated, >= 50% coverage\n",
       "2     3                                        wild type\n",
       "3     4               partial wild type, >= 50% coverage\n",
       "4     5  target and off-target mutation, >= 50% coverage\n",
       "5     6             off-target mutation, >= 50% coverage\n",
       "6     7                        truncated, < 50% coverage\n",
       "7    99                                         no reads"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "seq_confirm_code_df = pd.read_table(f\"{allele_collection_inputs}/seq_confirmation_results/sequence_confirmation_class_code.tsv\")\n",
    "with pd.option_context('display.max_columns', None):\n",
    "    display(seq_confirm_code_df)\n",
    "# seq_confirm_res_df[\"db_plate\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>symbol</th>\n",
       "      <th>orf_id_wt</th>\n",
       "      <th>mutation_id_old</th>\n",
       "      <th>ccsb_mutation_id</th>\n",
       "      <th>nt_change</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>collection</th>\n",
       "      <th>entry_plate_orig</th>\n",
       "      <th>entry_well_orig</th>\n",
       "      <th>entry_plate_conso</th>\n",
       "      <th>entry_well_conso</th>\n",
       "      <th>entry_seq_pool</th>\n",
       "      <th>db_plate</th>\n",
       "      <th>db_well</th>\n",
       "      <th>n2h_plate</th>\n",
       "      <th>n2h_well</th>\n",
       "      <th>dualip_plate</th>\n",
       "      <th>dualip_well</th>\n",
       "      <th>entry_sequenced</th>\n",
       "      <th>entry_sequence_confirmation_class</th>\n",
       "      <th>db_sequenced</th>\n",
       "      <th>db_sequence_confirmation_class</th>\n",
       "      <th>n2h_sequenced</th>\n",
       "      <th>n2h_sequence_confirmation_class</th>\n",
       "      <th>dualip_sequenced</th>\n",
       "      <th>dualip_sequence_confirmation_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GBA</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>CCSBVarC000001</td>\n",
       "      <td>160G&gt;C</td>\n",
       "      <td>Val54Leu</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>H01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>F12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>H01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GBA</td>\n",
       "      <td>2</td>\n",
       "      <td>73</td>\n",
       "      <td>CCSBVarC000002</td>\n",
       "      <td>670C&gt;T</td>\n",
       "      <td>Leu224Phe</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>E01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>C12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>E01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GBA</td>\n",
       "      <td>2</td>\n",
       "      <td>113</td>\n",
       "      <td>CCSBVarC000003</td>\n",
       "      <td>887G&gt;A</td>\n",
       "      <td>Arg296Gln</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>F01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>D12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>F01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GBA</td>\n",
       "      <td>2</td>\n",
       "      <td>231</td>\n",
       "      <td>CCSBVarC000004</td>\n",
       "      <td>1448T&gt;C</td>\n",
       "      <td>Leu483Pro</td>\n",
       "      <td>RC4</td>\n",
       "      <td>RC4_Mut_GDEh1026</td>\n",
       "      <td>G01</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>E12</td>\n",
       "      <td>2.0</td>\n",
       "      <td>RC4_Mut_GDDh1026</td>\n",
       "      <td>G01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GBA</td>\n",
       "      <td>2</td>\n",
       "      <td>213510</td>\n",
       "      <td>CCSBVarC003869</td>\n",
       "      <td>259C&gt;T</td>\n",
       "      <td>Arg87Trp</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1035</td>\n",
       "      <td>B03</td>\n",
       "      <td>GDEhDisVCh_40054</td>\n",
       "      <td>B02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CegsMutGDDh1035</td>\n",
       "      <td>B03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8705</th>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>100070227</td>\n",
       "      <td>212487</td>\n",
       "      <td>CCSBVarC007073</td>\n",
       "      <td>860A&gt;G</td>\n",
       "      <td>Asn287Ser</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>B07</td>\n",
       "      <td>GDEhDisVCh_40034</td>\n",
       "      <td>E09</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8706</th>\n",
       "      <td>TBX19</td>\n",
       "      <td>100070258</td>\n",
       "      <td>212488</td>\n",
       "      <td>CCSBVarC007074</td>\n",
       "      <td>383C&gt;T</td>\n",
       "      <td>Ser128Phe</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1043</td>\n",
       "      <td>A07</td>\n",
       "      <td>GDEhDisVCh_40046</td>\n",
       "      <td>B02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CegsMutGDDh1043</td>\n",
       "      <td>A07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8707</th>\n",
       "      <td>TBX19</td>\n",
       "      <td>100070258</td>\n",
       "      <td>212489</td>\n",
       "      <td>CCSBVarC007075</td>\n",
       "      <td>257T&gt;G</td>\n",
       "      <td>Met86Arg</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1043</td>\n",
       "      <td>G08</td>\n",
       "      <td>GDEhDisVCh_40046</td>\n",
       "      <td>F02</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CegsMutGDDh1043</td>\n",
       "      <td>G08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8708</th>\n",
       "      <td>GDF2</td>\n",
       "      <td>100070273</td>\n",
       "      <td>212491</td>\n",
       "      <td>CCSBVarC007076</td>\n",
       "      <td>203G&gt;T</td>\n",
       "      <td>Arg68Leu</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>E04</td>\n",
       "      <td>GDEhDisVCh_40033</td>\n",
       "      <td>B08</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8709</th>\n",
       "      <td>GDF2</td>\n",
       "      <td>100070273</td>\n",
       "      <td>212492</td>\n",
       "      <td>CCSBVarC007077</td>\n",
       "      <td>997C&gt;T</td>\n",
       "      <td>Arg333Trp</td>\n",
       "      <td>CEGS2</td>\n",
       "      <td>CegsMutGDEh1023</td>\n",
       "      <td>B06</td>\n",
       "      <td>GDEhDisVCh_40033</td>\n",
       "      <td>E08</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8710 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      symbol  orf_id_wt  mutation_id_old ccsb_mutation_id nt_change  \\\n",
       "0        GBA          2                6   CCSBVarC000001    160G>C   \n",
       "1        GBA          2               73   CCSBVarC000002    670C>T   \n",
       "2        GBA          2              113   CCSBVarC000003    887G>A   \n",
       "3        GBA          2              231   CCSBVarC000004   1448T>C   \n",
       "4        GBA          2           213510   CCSBVarC003869    259C>T   \n",
       "...      ...        ...              ...              ...       ...   \n",
       "8705  CTNNB1  100070227           212487   CCSBVarC007073    860A>G   \n",
       "8706   TBX19  100070258           212488   CCSBVarC007074    383C>T   \n",
       "8707   TBX19  100070258           212489   CCSBVarC007075    257T>G   \n",
       "8708    GDF2  100070273           212491   CCSBVarC007076    203G>T   \n",
       "8709    GDF2  100070273           212492   CCSBVarC007077    997C>T   \n",
       "\n",
       "      aa_change collection  entry_plate_orig entry_well_orig  \\\n",
       "0      Val54Leu        RC4  RC4_Mut_GDEh1026             H01   \n",
       "1     Leu224Phe        RC4  RC4_Mut_GDEh1026             E01   \n",
       "2     Arg296Gln        RC4  RC4_Mut_GDEh1026             F01   \n",
       "3     Leu483Pro        RC4  RC4_Mut_GDEh1026             G01   \n",
       "4      Arg87Trp      CEGS2   CegsMutGDEh1035             B03   \n",
       "...         ...        ...               ...             ...   \n",
       "8705  Asn287Ser      CEGS2   CegsMutGDEh1023             B07   \n",
       "8706  Ser128Phe      CEGS2   CegsMutGDEh1043             A07   \n",
       "8707   Met86Arg      CEGS2   CegsMutGDEh1043             G08   \n",
       "8708   Arg68Leu      CEGS2   CegsMutGDEh1023             E04   \n",
       "8709  Arg333Trp      CEGS2   CegsMutGDEh1023             B06   \n",
       "\n",
       "     entry_plate_conso entry_well_conso  entry_seq_pool          db_plate  \\\n",
       "0     GDEhDisVCh_40054              F12             2.0  RC4_Mut_GDDh1026   \n",
       "1     GDEhDisVCh_40054              C12             2.0  RC4_Mut_GDDh1026   \n",
       "2     GDEhDisVCh_40054              D12             2.0  RC4_Mut_GDDh1026   \n",
       "3     GDEhDisVCh_40054              E12             2.0  RC4_Mut_GDDh1026   \n",
       "4     GDEhDisVCh_40054              B02             2.0   CegsMutGDDh1035   \n",
       "...                ...              ...             ...               ...   \n",
       "8705  GDEhDisVCh_40034              E09             2.0               NaN   \n",
       "8706  GDEhDisVCh_40046              B02             2.0   CegsMutGDDh1043   \n",
       "8707  GDEhDisVCh_40046              F02             2.0   CegsMutGDDh1043   \n",
       "8708  GDEhDisVCh_40033              B08             1.0               NaN   \n",
       "8709  GDEhDisVCh_40033              E08             1.0               NaN   \n",
       "\n",
       "     db_well n2h_plate n2h_well dualip_plate dualip_well  entry_sequenced  \\\n",
       "0        H01       NaN      NaN          NaN         NaN                1   \n",
       "1        E01       NaN      NaN          NaN         NaN                1   \n",
       "2        F01       NaN      NaN          NaN         NaN                1   \n",
       "3        G01       NaN      NaN          NaN         NaN                1   \n",
       "4        B03       NaN      NaN          NaN         NaN                1   \n",
       "...      ...       ...      ...          ...         ...              ...   \n",
       "8705     NaN       NaN      NaN          NaN         NaN                1   \n",
       "8706     A07       NaN      NaN          NaN         NaN                1   \n",
       "8707     G08       NaN      NaN          NaN         NaN                1   \n",
       "8708     NaN       NaN      NaN          NaN         NaN                1   \n",
       "8709     NaN       NaN      NaN          NaN         NaN                1   \n",
       "\n",
       "      entry_sequence_confirmation_class  db_sequenced  \\\n",
       "0                                   1.0             1   \n",
       "1                                   1.0             1   \n",
       "2                                   1.0             1   \n",
       "3                                   1.0             1   \n",
       "4                                   1.0             1   \n",
       "...                                 ...           ...   \n",
       "8705                                1.0             0   \n",
       "8706                                7.0             1   \n",
       "8707                                1.0             1   \n",
       "8708                                5.0             0   \n",
       "8709                                5.0             0   \n",
       "\n",
       "      db_sequence_confirmation_class  n2h_sequenced  \\\n",
       "0                                2.0              0   \n",
       "1                                1.0              0   \n",
       "2                                7.0              0   \n",
       "3                                2.0              0   \n",
       "4                                2.0              0   \n",
       "...                              ...            ...   \n",
       "8705                             NaN              0   \n",
       "8706                             7.0              0   \n",
       "8707                             1.0              0   \n",
       "8708                             NaN              0   \n",
       "8709                             NaN              0   \n",
       "\n",
       "      n2h_sequence_confirmation_class  dualip_sequenced  \\\n",
       "0                                 NaN                 0   \n",
       "1                                 NaN                 0   \n",
       "2                                 NaN                 0   \n",
       "3                                 NaN                 0   \n",
       "4                                 NaN                 0   \n",
       "...                               ...               ...   \n",
       "8705                              NaN                 0   \n",
       "8706                              NaN                 0   \n",
       "8707                              NaN                 0   \n",
       "8708                              NaN                 0   \n",
       "8709                              NaN                 0   \n",
       "\n",
       "      dualip_sequence_confirmation_class  \n",
       "0                                    NaN  \n",
       "1                                    NaN  \n",
       "2                                    NaN  \n",
       "3                                    NaN  \n",
       "4                                    NaN  \n",
       "...                                  ...  \n",
       "8705                                 NaN  \n",
       "8706                                 NaN  \n",
       "8707                                 NaN  \n",
       "8708                                 NaN  \n",
       "8709                                 NaN  \n",
       "\n",
       "[8710 rows x 26 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "array([nan, 'VUSMutpDEST2_01', 'VUSMutpDEST2_02', 'VUSMutpDEST2_03',\n",
       "       'VUSMutpDEST2_04', 'VUSMutpDEST2_05', 'VUSMutpDEST2_06',\n",
       "       'VUSMutpDEST2_07', 'VUSMutpDEST2_08', 'VUSMutpDEST2_09',\n",
       "       'VUSMutpDEST2_11', 'VUSMutpDEST2_12', 'VUSMutpDEST2_13',\n",
       "       'VUSMutpDEST2_14', 'VUSMutpDEST2_15', 'VUSMutpDEST2_16'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_confirm_res_df = pd.read_table(f\"{allele_collection_inputs}/seq_confirmation_results/VarChampSeqConfirmationResult.tsv\")\n",
    "with pd.option_context('display.max_columns', None):\n",
    "    display(seq_confirm_res_df)\n",
    "seq_confirm_res_df[\"dualip_plate\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdest_layout_seq_conf_df = pd.merge(pdest_layout_df, \n",
    "        seq_confirm_res_df[\n",
    "        [\"symbol\",\"orf_id_wt\",\"mutation_id_old\",\"dualip_sequenced\",\"dualip_sequence_confirmation_class\",\"aa_change\",\"nt_change\",\"ccsb_mutation_id\"]\n",
    "        ].rename({\"orf_id_wt\": \"orf_id\",\n",
    "                \"mutation_id_old\": \"mut_id\"}, axis=1), \n",
    "    on=[\"orf_id\",\"mut_id\"],\n",
    "    how=\"left\"\n",
    ")\n",
    "# print(pdest_layout_seq_conf_df[\"dualip_sequence_confirmation_class\"].unique())\n",
    "\n",
    "## QC with allele collection df for consistency\n",
    "meta_cols = ['orf_id', 'mut_id', 'aa_change', 'nt_change', 'symbol']\n",
    "pdest_layout_seq_conf_df = pdest_layout_seq_conf_df.merge(allele_info_df[meta_cols], on=['orf_id', 'mut_id'], how='left', suffixes=(\"\", \"_ac\"))\n",
    "pdest_layout_seq_conf_df\n",
    "\n",
    "assert(all(pdest_layout_seq_conf_df.dropna(subset=\"symbol\")[\"symbol\"]==pdest_layout_seq_conf_df.dropna(subset=\"symbol\")[\"symbol_ac\"]))\n",
    "assert(all(pdest_layout_seq_conf_df.dropna(subset=\"aa_change\")[\"aa_change\"]==pdest_layout_seq_conf_df.dropna(subset=\"aa_change\")[\"aa_change_ac\"]))\n",
    "assert(all(pdest_layout_seq_conf_df.dropna(subset=\"nt_change\")[\"nt_change\"]==pdest_layout_seq_conf_df.dropna(subset=\"nt_change\")[\"nt_change_ac\"]))\n",
    "pdest_layout_seq_conf_df = pdest_layout_seq_conf_df.drop(columns=[\"symbol_ac\",\"aa_change_ac\",\"nt_change_ac\"])\n",
    "\n",
    "## replace NAs\n",
    "pdest_layout_seq_conf_df[\"orf_id\"] = pdest_layout_seq_conf_df[\"orf_id\"].astype(\"Int64\").astype(str)\n",
    "pdest_layout_seq_conf_df[\"orf_id\"] = pdest_layout_seq_conf_df[\"orf_id\"].replace(\"<NA>\", np.nan)\n",
    "pdest_layout_seq_conf_df[[\"aa_change\", \"nt_change\"]] = pdest_layout_seq_conf_df[[\"aa_change\", \"nt_change\"]].fillna(\"WT\")\n",
    "## Replace missing symbols\n",
    "pdest_layout_seq_conf_df[\"symbol\"] = pdest_layout_seq_conf_df[\"symbol\"].fillna(pdest_layout_seq_conf_df[\"orf_id\"].map(orf_to_symbol))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdest_layout_seq_conf_df.to_csv(f'{dualipa_outputs}/dualipa_experimental_layout_seq_conf.csv', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Add metadata\n",
    "# meta_cols = ['orf_id', 'mut_id', 'aa_change', 'nt_change', 'symbol']\n",
    "# pdest_layout_df = pdest_layout_df.merge(allele_info_df[meta_cols], on=['orf_id', 'mut_id'], how='left')\n",
    "# pdest_layout_df[[\"aa_change\", \"nt_change\"]] = pdest_layout_df[[\"aa_change\", \"nt_change\"]].fillna(\"WT\")\n",
    "# pdest_layout_df[\"orf_id\"] = pdest_layout_df[\"orf_id\"].astype(\"Int64\").astype(str)\n",
    "# pdest_layout_df[\"orf_id\"] = pdest_layout_df[\"orf_id\"].replace(\"<NA>\", np.nan)\n",
    "\n",
    "# # Replace missing symbols\n",
    "# pdest_layout_df[\"symbol\"] = pdest_layout_df[\"symbol\"].fillna(pdest_layout_df[\"orf_id\"].map(orf_to_symbol))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# pdest_layout_df.to_csv(f'{dualipa_outputs}/dualipa_experimental_layout.csv', index=None)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ai4protein",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
